File: clients/algoliasearch-client-php/lib/Model/Search/SearchForFacetsOptions.php
<?php

namespace Algolia\AlgoliaSearch\Model\Search;

/**
 * SearchForFacetsOptions Class Doc Comment
 *
 * @category Class
 * @package Algolia\AlgoliaSearch
 */
class SearchForFacetsOptions extends \Algolia\AlgoliaSearch\Model\AbstractModel implements
        ModelInterface,
        \ArrayAccess,
        \JsonSerializable
{
    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @var string[]
     */
    protected static $modelTypes = [
        'facet' => 'string',
        'indexName' => 'string',
        'facetQuery' => 'string',
        'maxFacetHits' => 'int',
        'type' => '\Algolia\AlgoliaSearch\Model\Search\SearchTypeFacet',
    ];

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @var string[]
     */
    protected static $modelFormats = [
        'facet' => null,
        'indexName' => null,
        'facetQuery' => null,
        'maxFacetHits' => null,
        'type' => null,
    ];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelTypes()
    {
        return self::$modelTypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function modelFormats()
    {
        return self::$modelFormats;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'facet' => 'setFacet',
        'indexName' => 'setIndexName',
        'facetQuery' => 'setFacetQuery',
        'maxFacetHits' => 'setMaxFacetHits',
        'type' => 'setType',
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'facet' => 'getFacet',
        'indexName' => 'getIndexName',
        'facetQuery' => 'getFacetQuery',
        'maxFacetHits' => 'getMaxFacetHits',
        'type' => 'getType',
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     */
    public function __construct(array $data = null)
    {
        if (isset($data['facet'])) {
            $this->container['facet'] = $data['facet'];
        }
        if (isset($data['indexName'])) {
            $this->container['indexName'] = $data['indexName'];
        }
        if (isset($data['facetQuery'])) {
            $this->container['facetQuery'] = $data['facetQuery'];
        }
        if (isset($data['maxFacetHits'])) {
            $this->container['maxFacetHits'] = $data['maxFacetHits'];
        }
        if (isset($data['type'])) {
            $this->container['type'] = $data['type'];
        }
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if (
            !isset($this->container['facet']) ||
            $this->container['facet'] === null
        ) {
            $invalidProperties[] = "'facet' can't be null";
        }
        if (
            !isset($this->container['indexName']) ||
            $this->container['indexName'] === null
        ) {
            $invalidProperties[] = "'indexName' can't be null";
        }
        if (
            isset($this->container['maxFacetHits']) &&
            $this->container['maxFacetHits'] > 100
        ) {
            $invalidProperties[] =
                "invalid value for 'maxFacetHits', must be smaller than or equal to 100.";
        }

        if (
            !isset($this->container['type']) ||
            $this->container['type'] === null
        ) {
            $invalidProperties[] = "'type' can't be null";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }

    /**
     * Gets facet
     *
     * @return string
     */
    public function getFacet()
    {
        return $this->container['facet'] ?? null;
    }

    /**
     * Sets facet
     *
     * @param string $facet the `facet` name
     *
     * @return self
     */
    public function setFacet($facet)
    {
        $this->container['facet'] = $facet;

        return $this;
    }

    /**
     * Gets indexName
     *
     * @return string
     */
    public function getIndexName()
    {
        return $this->container['indexName'] ?? null;
    }

    /**
     * Sets indexName
     *
     * @param string $indexName the Algolia index name
     *
     * @return self
     */
    public function setIndexName($indexName)
    {
        $this->container['indexName'] = $indexName;

        return $this;
    }

    /**
     * Gets facetQuery
     *
     * @return string|null
     */
    public function getFacetQuery()
    {
        return $this->container['facetQuery'] ?? null;
    }

    /**
     * Sets facetQuery
     *
     * @param string|null $facetQuery text to search inside the facet's values
     *
     * @return self
     */
    public function setFacetQuery($facetQuery)
    {
        $this->container['facetQuery'] = $facetQuery;

        return $this;
    }

    /**
     * Gets maxFacetHits
     *
     * @return int|null
     */
    public function getMaxFacetHits()
    {
        return $this->container['maxFacetHits'] ?? null;
    }

    /**
     * Sets maxFacetHits
     *
     * @param int|null $maxFacetHits Maximum number of facet hits to return during a search for facet values. For performance reasons, the maximum allowed number of returned values is 100.
     *
     * @return self
     */
    public function setMaxFacetHits($maxFacetHits)
    {
        if (!is_null($maxFacetHits) && $maxFacetHits > 100) {
            throw new \InvalidArgumentException(
                'invalid value for $maxFacetHits when calling SearchForFacetsOptions., must be smaller than or equal to 100.'
            );
        }

        $this->container['maxFacetHits'] = $maxFacetHits;

        return $this;
    }

    /**
     * Gets type
     *
     * @return \Algolia\AlgoliaSearch\Model\Search\SearchTypeFacet
     */
    public function getType()
    {
        return $this->container['type'] ?? null;
    }

    /**
     * Sets type
     *
     * @param \Algolia\AlgoliaSearch\Model\Search\SearchTypeFacet $type type
     *
     * @return self
     */
    public function setType($type)
    {
        $this->container['type'] = $type;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param int $offset Offset
     *
     * @return bool
     */
    public function offsetExists($offset)
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param int $offset Offset
     *
     * @return mixed|null
     */
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value)
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param int $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset)
    {
        unset($this->container[$offset]);
    }
}

File: clients/algoliasearch-client-java-2/algoliasearch-core/src/main/java/com/algolia/model/search/SearchForFacetsOptions.java
package com.algolia.model.search;

import com.fasterxml.jackson.annotation.*;
import java.util.Objects;

/** SearchForFacetsOptions */
public class SearchForFacetsOptions {

  @JsonProperty("facet")
  private String facet;

  @JsonProperty("indexName")
  private String indexName;

  @JsonProperty("facetQuery")
  private String facetQuery;

  @JsonProperty("maxFacetHits")
  private Integer maxFacetHits;

  @JsonProperty("type")
  private SearchTypeFacet type;

  public SearchForFacetsOptions setFacet(String facet) {
    this.facet = facet;
    return this;
  }

  /**
   * The `facet` name.
   *
   * @return facet
   */
  @javax.annotation.Nonnull
  public String getFacet() {
    return facet;
  }

  public SearchForFacetsOptions setIndexName(String indexName) {
    this.indexName = indexName;
    return this;
  }

  /**
   * The Algolia index name.
   *
   * @return indexName
   */
  @javax.annotation.Nonnull
  public String getIndexName() {
    return indexName;
  }

  public SearchForFacetsOptions setFacetQuery(String facetQuery) {
    this.facetQuery = facetQuery;
    return this;
  }

  /**
   * Text to search inside the facet's values.
   *
   * @return facetQuery
   */
  @javax.annotation.Nullable
  public String getFacetQuery() {
    return facetQuery;
  }

  public SearchForFacetsOptions setMaxFacetHits(Integer maxFacetHits) {
    this.maxFacetHits = maxFacetHits;
    return this;
  }

  /**
   * Maximum number of facet hits to return during a search for facet values. For performance
   * reasons, the maximum allowed number of returned values is 100. maximum: 100
   *
   * @return maxFacetHits
   */
  @javax.annotation.Nullable
  public Integer getMaxFacetHits() {
    return maxFacetHits;
  }

  public SearchForFacetsOptions setType(SearchTypeFacet type) {
    this.type = type;
    return this;
  }

  /**
   * Get type
   *
   * @return type
   */
  @javax.annotation.Nonnull
  public SearchTypeFacet getType() {
    return type;
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    SearchForFacetsOptions searchForFacetsOptions = (SearchForFacetsOptions) o;
    return (
      Objects.equals(this.facet, searchForFacetsOptions.facet) &&
      Objects.equals(this.indexName, searchForFacetsOptions.indexName) &&
      Objects.equals(this.facetQuery, searchForFacetsOptions.facetQuery) &&
      Objects.equals(this.maxFacetHits, searchForFacetsOptions.maxFacetHits) &&
      Objects.equals(this.type, searchForFacetsOptions.type)
    );
  }

  @Override
  public int hashCode() {
    return Objects.hash(facet, indexName, facetQuery, maxFacetHits, type);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class SearchForFacetsOptions {\n");
    sb.append("    facet: ").append(toIndentedString(facet)).append("\n");
    sb.append("    indexName: ").append(toIndentedString(indexName)).append("\n");
    sb.append("    facetQuery: ").append(toIndentedString(facetQuery)).append("\n");
    sb.append("    maxFacetHits: ").append(toIndentedString(maxFacetHits)).append("\n");
    sb.append("    type: ").append(toIndentedString(type)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}
